storage/pebble/receipts.go

```diff
@@ -19,13 +18,11 @@ var _ storage.ReceiptIndexer = &Receipts{}
 
 type Receipts struct {
 	store *Storage
-	mux   sync.RWMutex
 }
 
 func NewReceipts(store *Storage) *Receipts {
 	return &Receipts{
 		store: store,
-		mux:   sync.RWMutex{},
```

Comment:
How come we removed the `sync.RWMutex` from this type? Are we certain that `Receipts.Store` has finished writing all the necessary mappings for a single receipt, before `Receipts.GetByTransactionID` / `Receipts.BloomsForBlockRange` attempts to get it?

It seems that we consistently use a `RWMutex` on all storage indexes:
- `Blocks`
- `Transactions`
- `Receipts`
- `Traces`

Reply:
the locks are not necessary.
pebble can handle concurrent writes and reads.
If a batch has been committed the read will see new data, if not, it will see the old data.

a batch is not concurrency safe, you cant write/read from the same batch in multiple go-routines, but we never do that. Only bootstrapping and re-execution are creating and saving batches, and those are on a single goroutine.